corehq/apps/user_importer/tests/test_importer.py

```diff
@@ -1727,3 +1729,98 @@ def test_invite_location_add(self):
     def setup_locations(self):
         self.loc1 = make_loc('loc1', type='state', domain=self.domain_name)
         self.loc2 = make_loc('loc2', type='state', domain=self.domain_name)
+
+
+class TestUserChangeLogger(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_name = 'mydomain'
+        cls.domain = Domain.get_or_create_with_name(name=cls.domain_name)
+        cls.uploading_user = WebUser.create(cls.domain_name, "admin@xyz.com", 'password', None, None,
+                                            is_superuser=True)
+        cls.upload_record = UserUploadRecord(
+            domain=cls.domain_name,
+            user_id=cls.uploading_user.get_id
+        )
+        cls.upload_record.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.upload_record.delete()
+        cls.uploading_user.delete(cls.domain_name, deleted_by=None)
+        cls.domain.delete()
+        super().tearDownClass()
+
+    def test_add_change_message_duplicate_slug_entry(self):
+        new_user = CommCareUser()
+        user_change_logger = UserChangeLogger(
+            domain=self.domain_name,
+            user=new_user,
+            is_new_user=True,
+            changed_by_user=self.uploading_user,
+            changed_via=USER_CHANGE_VIA_BULK_IMPORTER,
+            upload_record_id=self.upload_record.pk
+        )
+        user_change_logger.add_change_message(UserChangeMessage.password_reset())
+
+        # no change noted for new user
+        self.assertEqual(user_change_logger.change_messages, {})
+
+        # no exception raised for new user
+        user_change_logger.add_change_message(UserChangeMessage.password_reset())
+
+        user = CommCareUser.create(self.domain_name, f"hello@{self.domain.name}.commcarehq.org", "*******",
+                                   created_by=None, created_via=None)
+        self.addCleanup(user.delete, self.domain_name, deleted_by=None)
```

Comment:
do actually need to save the user here? I think you can just pass in `is_new_user=False` with an unsaved user model can't you? Perhaps then you could make this a non-db test.